### PR TITLE
AKU-873: AlfMenuBarSelect (and related fixes)

### PR DESCRIPTION
**Pull request by draperd**
This PR addresses https://issues.alfresco.com/jira/browse/AKU-873 to ensure that icon updating works correctly with the AlfMenuBarSelect. However, along the way I've fixed up the basic menus tests and ensured that popup menus can share the same icon suite as other menu widgets.

### CODE REVIEW

File: aikau/src/main/resources/alfresco/menus/AlfMenuBarPopup.js [hunk 1]
```diff
@@ -109,32 +109,51 @@ define(["dojo/_base/declare",
...
             this.title = this.message(this.title);
          }
+         if (!this.iconAltText && this.label)
+         {
+            this.iconAltText = this.label;
+         }
+         else
+         {
+            this.iconAltText = this.message(this.iconAltText);
+         }
```
Comment (MartinDoyleUK) [inline aikau/src/main/resources/alfresco/menus/AlfMenuBarPopup.js]: It looks like `this.label` won't be "messaged"? Could this just be written as `this.iconAltText = this.message(this.iconAltText || this.label);`?

File: aikau/src/main/resources/alfresco/menus/_AlfMenuItemMixin.js [hunk 2]
```diff
@@ -148,13 +149,24 @@ define(["dojo/_base/declare",
       postMixInProperties: function alfresco_menus__AlfMenuItemMixin__postMixInProperties() {
          if (this.label)
          {
-            this.params.label = this.label = this.encodeHTML(this.message(this.label));
+            this.params.label = this.label = this.message(this.label);
          }
-         if (this.title)
+         if (!this.title)
+         {
+            this.title = this.label;
```
Comment (MartinDoyleUK) [inline aikau/src/main/resources/alfresco/menus/_AlfMenuItemMixin.js]: Again, a possible non "messaging"??

File: aikau/src/main/resources/alfresco/menus/AlfMenuBarPopup.js (hunk 1 above)
Comment (draperd) [inline aikau/src/main/resources/alfresco/menus/AlfMenuBarPopup.js]: It's not obvious from how Git shows the diff, but this.label has already been "messaged" directly before, so if it's available it's the messaged form.

File: aikau/src/main/resources/alfresco/menus/_AlfMenuItemMixin.js (hunk 2 above)
Comment (draperd) [inline aikau/src/main/resources/alfresco/menus/_AlfMenuItemMixin.js]: As before, except I'm missing the check that label is set

**Pull request merged**